hbase-common/src/main/java/org/apache/hadoop/hbase/util/UnsafeAvailChecker.java

```diff
@@ -33,12 +35,16 @@
   private static final Logger LOG = LoggerFactory.getLogger(UnsafeAvailChecker.class);
   private static boolean avail = false;
   private static boolean unaligned = false;
+  // Flag that can be set by clients to enable/disable Unsafe package usage
+  private static boolean useUnsafe;
```

Comment:
Could we make these flags final? If they are final, then JIT could optimize them soon...

Reply:
@Apache9 We might override the useUnsafe flag to enable/disable the Unsafe package usage on the client side using overrideUnsafeAvail method

Reply:
Could we make use of a environment variable to enable or disable the unsafe usage? I do not expect we should support switching the unsafe usage on the fly

Reply:
@Apache9 Yes that is definitely a good idea and in fact we have started the change  with having a configuration parameter that can be set to false for disabling the Unsafe usage which would be read in the static block ( of UnsafeAvailChecker) for not having the clients to explicitly call a method to disable the Unsafe availability 
@apurtell 
But one thing that we observed is that not all clients have the hbase-site or hbase-default xml in the classpaths where they can set this flag . Hence we thought of having a util method (in addition to the configuration way of disabling it) for helping those clients to disable the Unsafe usage  . Please provide your thoughts on any other way that we can help achieve this.

Reply:
I do not think adding configs to hbase-site.xml or hbase-default.xml can help here, as if we want to make it final then we need to initialize these flags in the static initialization section, where we do not pass in a Configuration.
So I suggest we use a environment variable to control the behavior, i.e, add `-Dorg.apache.hbase.unsafe=false` in the start command and we will test this env variable in the static initialization section to determine whether to disable unsafe.

Reply:
@Apache9  I have made the PR simple and removed the code for replacing static with non-static references in Bytes class. Also I have removed the useUnsafe variable and just utilizing the avail variable based on the set config. Please review